### PR TITLE
Cleanup server addparticle(spawner) by merge two identical functions.

### PR DESCRIPTION
**Pull request by nerzhul**

### CODE REVIEW

File: src/server.cpp [hunk 1]
```diff
@@ -3146,26 +3119,21 @@ u32 Server::addParticleSpawnerAll(u16 amount, float spawntime,
 	return id;
 }
 
-void Server::deleteParticleSpawner(const char *playername, u32 id)
+void Server::deleteParticleSpawner(std::string playername, u32 id)
```
Comment (est31) [inline src/server.cpp]: Don't do this! This copies the entire string! Do this instead: `const std::string &playername`

**Comment by est31** [comment]
OK except std::string copy problem

**Comment by nerzhul** [comment]
std::string fixed. is  that okay ?

**Comment by est31** [comment]
No, one std::string is still there.

**Comment by est31** [comment]
:+1:

**Pull request merged**